src/models/userDao.js

```diff
@@ -0,0 +1,44 @@
+const dataSource = require('./data-source')
+
+const getUserByKakaoId = async (kakaoId) => {
+    try{
+        const result = await dataSource.query(`
+            SELECT 
+                kakao_id
+            FROM
+                users
+            WHERE
+                kakao_id = ${kakaoId}
+            `)
+
+        return result.length
+    } catch {
+        throw new Error('getUserByKakaoIdErr')
+    }
+}
+
+const insertKakaoId = async (kakaoId, email, profileImage, nickname) => {
+    try {
+        const result = await dataSource.query(`
+            INSERT INTO users (
+                kakao_id,
+                email,
+                profile_image,
+                nickname
+            ) VALUES (
+                ?,
+                ?,
+                ?,
+                ?
+            )`,
+            [kakaoId, email, profileImage, nickname]
+        )
```

Comment:
```suggestion
    try {
        return await dataSource.query(`
            INSERT INTO users (
                kakao_id,
                email,
                profile_image,
                nickname
            ) VALUES (
                ?,
                ?,
                ?,
                ?
            )`,
            [kakaoId, email, profileImage, nickname]
        )
```
속해있는 scope에서 result라는 변수가 쓰이지 않고 있기 때문에 선언을 하지 않아도 될 것 같습니다.